pkg/noun/events.c

```diff
@@ -398,23 +398,31 @@ _ce_ephemeral_open(c3_i* eph_i)
 /* _ce_image_open(): open or create image.
 */
 static c3_o
-_ce_image_open(u3e_image* img_u)
+_ce_image_open(u3e_image* img_u, c3_c* ful_c)
 {
   c3_i mod_i = O_RDWR | O_CREAT;
-  c3_c ful_c[8193];
-
-  snprintf(ful_c, 8192, "%s", u3P.dir_c);
-  c3_mkdir(ful_c, 0700);
 
-  snprintf(ful_c, 8192, "%s/.urb", u3P.dir_c);
-  c3_mkdir(ful_c, 0700);
-
-  snprintf(ful_c, 8192, "%s/.urb/chk", u3P.dir_c);
-  c3_mkdir(ful_c, 0700);
+  //  create directories recursively like `mkdir -p`
+  //
+  {
+    c3_c* dir_c = ful_c;
+    c3_c* end_c = ful_c + strlen(ful_c);
+
+    while ( dir_c < end_c ) {
+      if ( '/' == *dir_c ) {
+        *dir_c = 0;
+        c3_mkdir(ful_c, 0700);
+        *dir_c = '/';
+      }
+      dir_c++;
+    }
+    c3_mkdir(ful_c, 0700);
+  }
```

Comment:
compared to the previous change, this should correctly create the parent directories, but I think it's wholly unnecessary and also this "traverse path and replace / with \0, mkdir path, re-replace \0 with /" is just kinda janky and more generalized than we need.

rather than call _ce_image_open on a path + file that may not exist, we should just ensure the required parent directories already exist.

Those would be "./urb/log" (this necessarily must already exist) and "./urb/log/0iN" -- this WILL need to be created. An explicit mkdir 0iN would certainly be more readable.

Reply:
i.e. *retain* the behavior of creating images (files) that do not exist, but don't add extra behavior that creates parent directories if those also do not exist.

Reply:
Also `.urb/chk`, which is actually the only time we actually create the directories in `_ce_image_open` at the moment.

Reply:
Fair enough proposal to require the caller to ensure the parent directories exist before calling `_ce_image_open`-- I will implement it today.